netmiko/ipinfusion/ipinfusion_ocnos.py

```diff
@@ -0,0 +1,84 @@
+from __future__ import unicode_literals
+import time
+import re
+from telnetlib import IAC, DO, DONT, WILL, WONT, SB, SE, TTYPE
+from netmiko.ssh_exception import NetMikoTimeoutException
+from netmiko.cisco_base_connection import CiscoBaseConnection
+
+
+class IpinfusionOcnosBase(CiscoBaseConnection):
+    """Common Methods for IP Infusion OcNOS support."""
+    def session_preparation(self):
+        self._test_channel_read()
+        self.set_base_prompt()
+        self.disable_paging(command="terminal length 0")
+
+        # Clear the read buffer
+        time.sleep(.3 * self.global_delay_factor)
+        self.clear_buffer()
+
+    def save_config(self, cmd='write', confirm=False, confirm_response=''):
+        """Saves Config Using write command"""
+        return super(IpinfusionOcnosBase, self).save_config(
+            cmd=cmd, confirm=confirm, confirm_response=confirm_response)
+
+    def enable(self, cmd='enable', pattern='ssword', re_flags=re.IGNORECASE):
+        """Enter enable mode.
+
+        :param cmd: Device command to enter enable mode
+        :type cmd: str
+
+        :param pattern: pattern to search for indicating device is waiting for password
+        :type pattern: str
+
+        :param re_flags: Regular expression flags used in conjunction with pattern
+        :type re_flags: int
+        """
+        output = ""
+        msg = "Failed to enter enable mode. Please ensure you pass " \
+              "the 'secret' argument to ConnectHandler."
+        if not self.check_enable_mode():
+            self.write_channel(self.normalize_cmd(cmd))
+            try:
+                output += self.read_until_prompt_or_pattern(pattern=pattern, re_flags=re_flags)
+                self.write_channel(self.secret + self.TELNET_RETURN)
```

Comment:
I assume that we need to do this so that the standard <enter> is "\r\n"? 

If this is correct, let me know and I will make a slight modification that should work in a more consistent way with Netmiko.

Reply:
Sorry for asking twice on this as I now see you made note of this in the original comments. I think it is actually the device that is different here not Netmiko. I suspect the issue is that in the `enable` case the device is interpreting the \n as part of the password, but with \r\n it understands that it is the <enter>. Definite assumptions here on my part.

I have seen something similar before. I think for Aruba (IIRC).